### ISSUE TITLE
(SKCanvasView) Clear with Transparent does not work on XF/iOS. (also set Background as Transparent)

### ISSUE THREAD

**Issue by benevbright** [mono/SkiaSharp#281]
Hi Matthew. 

I alway want to say thanks to you and SkiaSharp. 
It helped us a lot. 

I have a problem to clear SKCanvasView with Transparent color on XF/iOS.
It's working correctly on XF/Android.
I'm using Skiasharp.forms. 
And I also set SKCanvasView's background color as Transparent.

I wonder if you could quick fix for this?

Thanks.

**Comment by mattleibow** [comment]
I have not been able to repro this. Can you attach a test case or some code?

Here is my sample: https://github.com/mattleibow/SkiaSharpXamarinFormsDemo

To demonstrate the transparency, I set a background color for the root `StackLayout` in the `MainPage.xaml`:

```xml
<StackLayout ... BackgroundColor="Fuchsia">
```

**Comment by benevbright** [comment]
Hi, 
Did you also try to redraw? 
It does not clear the previous bitmap when I need to clear previous one.

I will make repro project and attach it.

**Comment by benevbright** [comment]
Hm. It DOES clear the view on the sample repro project I made.
I will try it on the ViewCell of ListView again and will share the result here.

**Comment by benevbright** [comment]
Yes. I could reproduce it on the ViewCell. 
Don't know why this makes different behavior.

**Comment by benevbright** [comment]
[test_skiasharp_clearTransparent.zip](https://github.com/mono/SkiaSharp/files/962808/test_skiasharp_clearTransparent.zip)

Hi, Matthew. 

I made repro project. 
tested on iOS.
It happens when I use SKCanvasView on the ViewCell. 
If you have any workaround, please let me know.
Thanks.

It's working fine on XF/Android.

**Comment by benevbright** [comment]
I understand that this could be XF's problem.

**Comment by mattleibow** [comment]
I had a look, and I think the issue in the attached sample may be with the fact that the cell *is* transparent, but we are seeing the *background of the list view*. The list view has a white background by default.

If you want the page to show through, you first have to set the list view's `BackgroundColor` to `Transparent`.

Let me know if this solves it for you, or if this is not the issue you are having.

**Comment by mattleibow** [comment]
I found the issue. Thanks for reporting this. There is a bug in the Xamarin.Forms renderer where it is adding the native view multiple times when you start to scroll.

This will be fixed in the next release, but in the meantime a solution is to use the recycle the elements of the list using the `CachingStrategy="RecycleElement"` property in the XAML. Not only is this generally a good idea as the number of views is less overall, but it will also allow for smoother scrolling. 

You could use a cell like this:

```csharp
public class MyViewCell : ViewCell
{
    private SKCanvasView skiaview;

    public MyViewCell()
    {
        // crate the canvas
        skiaview = new SKCanvasView();

        // do the drawing
        skiaview.PaintSurface += (object sender, SKPaintSurfaceEventArgs e) =>
        {
        };

        // assign the canvas to the cell
        View = skiaview;
    }

    protected override void OnBindingContextChanged()
    {
        base.OnBindingContextChanged();

        // update the canvas when the data changes
        skiaview.InvalidateSurface();
    }
}
```

Does this help?

**Comment by benevbright** [comment]
Thanks Matthew, It works like a charm.
Thanks again for your kind help.

**Comment by mattleibow** [comment]
Fixed in commit: a2a4576597c53a17330b548b7f5cdb0211e20393

**Comment by harshpatel2810** [comment]
@mattleibow can i make canvas view with smooth scrolling.
i'm using skia skcanvasview but as i told that is not scrolling smoothly so if you can please help me with this.

Thanks...:)
